interface/src/app/tribe_auth/resource.js

```diff
@@ -0,0 +1,14 @@
+/*
+ * The resources for fetching the OAuth2 token from the tribe_client url
+ */
+angular.module("adage.tribe_auth.resource", ['ngResource'])
+    .factory('User', ['$resource', function($resource) {
+        return $resource('/tribe_client/return_user', {}, {
```

Comment:
Great suggestion, @dongbohu. This idea would simplify the code here. I think I have some instances that would benefit from that change as well.